pycbc/detector.py

```diff
@@ -444,3 +445,138 @@ def time_delay_from_earth_center(self, right_ascension, declination, t_gps):
+        ----------
+        ref_time : numpy.ScalarType
+            GPS time
+        hp, hc : pycbc.types.TimeSeries
+            Plus and cross polarizations in order
+        time : numpy.ndarray (3,1)
+            GPS time
+        Returns
+        -------
+        numpy.ndarray shape (3,1)
+            The values of hij at the required time
+        """
+        fhp = interpolate.interpolate.interp1d(hp.sample_times, hp)
+        fhc = interpolate.interpolate.interp1d(hc.sample_times, hc)
+        xnew = np.linspace(hp.sample_times[0], hp.sample_times[-1], len(hp)*100)
+        hp, hc = fhp(xnew), fhc(xnew)
+        fhp = TimeSeries(hp, delta_t=1/len(hp))
+        fhc = TimeSeries(hc, delta_t=1/len(hc))
+        return fhp, fhc, xnew
+
+    def eval_hij(self, hp, hc, k, ref_time, t):
+        """Returns the values of hij
+        Parameters
+        ----------
+        hp, hc : pycbc.types.TimeSeries
+            Plus and cross polarizations in order
+        k : numpy.ndarray shape(3,1)
+            Sky-loaction in Cartesian system
+        ref_time : numpy.ScalarType
+            GPS time
+        time : numpy.ScalarType
+            GPS time
+        Returns
+        -------
+        numpy.ndarray shape (3,1)
+            The values of hij at the required time
+        """
+        k = k/(k[0]**2 + k[1]**2 + k[2]**2)**.5
+        beta = np.arcsin(-k[2])
+        theta = np.pi/2 - beta
+        lam = np.arcsin(-k[1]/cos(beta))
+        e_theta = np.array([sin(beta)*cos(lam), sin(beta)*sin(lam), -cos(beta)])
+        e_phi = np.array([-sin(lam), cos(lam), 0])
+        u = np.copy(-e_phi)
+        v = np.copy(-e_theta)
+        phi = np.copy(lam)
+        psi = np.arctan((-sin(beta)*sin(theta)*cos(lam - phi) + cos(theta)*cos(beta)) / (sin(theta)*sin(lam-phi)))
+        epij = np.outer(u, u) - np.outer(v, v)
+        ecij = np.outer(u, v) + np.outer(v, u)
+        hp, hc, sample_time = self.interpolate_hij(hp, hc)
+        h = []
+        for i in range(len(t)):
+            t0 = np.abs(sample_time - t[i]).argmin()
+            hij = np.zeros((3, 3))
+            hij[0, 0:2] = (hp[t0]*cos(2*psi) - hc[t0]*sin(2*psi))*epij[0][0:2] + (hp[t0]*sin(2*psi) - hc[t0]*cos(2*psi))*ecij[0][0:2]
+            hij[1, 0:2] = (hp[t0]*cos(2*psi) - hc[t0]*sin(2*psi))*epij[1][1:3] + (hp[t0]*sin(2*psi) - hc[t0]*cos(2*psi))*ecij[1][1:3]
+            h.append(hij)
+        return np.array(h)
+
+    def GWresponse(self, ref_time, hp, hc, k):
+        """Returns the response to the GW signals
+        Parameters
+        ----------
+        ref_time : numpy.ScalarType
+            GPS time
+        hp, hc : pycbc.types.TimeSeries
+            Plus and cross polarizations in order
+        k : numpy.ndarray shape (3,1)
+            sky-location of the signal
+        Returns
+        -------
+        numpy.ndarray
+            The values of response to GW signals.
+        """
+        guide_cen = np.transpose(np.mean(self.get_pos(ref_time), axis=1))
```

Comment:
There shouldn't be a reference time here, you really should evaluate this at each time you need for the waveform given. If the signal lasts for any significant portion of a year for example, this won't give the correct result as the position will have significant shifted from start to end.